docs/preset-env.md

```diff
@@ -134,7 +134,7 @@ We recognize this isn’t ideal and will be revisiting this in Babel v8.
 
 You may also target browsers supporting ES Modules (https://www.ecma-international.org/ecma-262/6.0/#sec-modules). When specifying this option, the browsers field will be ignored. You can use this approach in combination with `<script type="module"></script>` to conditionally serve smaller scripts to users (https://jakearchibald.com/2017/es-modules-in-browsers/#nomodule-for-backwards-compatibility).
 
-> _Please note_: when specifying the esmodules target, browsers targets will be ignored.
+> _Please note_: when specifying the esmodules target, browsers targets will be ignored. But `browserslist`'s configuration will still work intersectly at the same time.
```

Comment:
Howabout:

```suggestion
> _Please note_: when the `esmodules` target is specified, the `browsers` target will be ignored. It _will_, however, intersect with `browserslist`'s targets.
```

Should we remove the similar statement in the paragraph above?

Reply:
Actually, there is a difference between the top-level `targets` option and the `targets` option of `@babe/preset-env` (we'll align them by changing the `@babel/preset-env` behavior in Babel 8).

- In the `@babel/preset-env` options, `esmodules: true` always overrides anything else: both the `browsers` options and the `.browserslistrc` file
- In the top-level `targets` option (https://babeljs.io/docs/en/options#targets), it will intersect the `browsers` option or `.browserlistrc` file.

However, I just noticed that https://babeljs.io/docs/en/options#targetsesmodules is wrong because it says that `browsers` will be ignored, while it's intersected.

Reply:
@nicolo-ribaudo I've changed `options.md` at the same time, please review again.